### PR TITLE
Pull Request for OOP school library: set up associations

### PR DESCRIPTION
**Pull request by chuaindia**
In this branch following activities are incorporated :

-  3 classes "Classroom", "Book" and "Rental" are created and the associations between them are established.
- The has-many/belongs-to relationship between Classroom and Student is created. 
- In the class "Book" instance variables  @title and @author  are created and initialized in the constructor.
- Setters and getters for instance variables are created with the "attr_accessor" option.
- The many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental is 
   established.
- The has-many side of Book and Rental (a book has many rentals) is created.
- The belongs-to side of Rental and Book (a rental belongs to a book) is created.
- The has-many side of Person and Rental (a person has many rentals) is created.
- The belongs-to side of Rental and Person (a rental belongs to a person) is created.
- The constructor of Rental is modified so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (mckent05) [inline book.rb]: - Please implement a method to add rentals, the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. Please fix this 👍 👍

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  def correct_name
+    @name
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+end
```
Comment (mckent05) [inline person.rb]: - Please implement a method to add rentals, the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association. Please fix this 👍 👍

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (mckent05) [inline rental.rb]: - According to the project requirements, the `:date,:book, :person` attributes should have getter and setter methods, please fix this 👍

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @chuaindia, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job using the right workflow for this milestone 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 3 above)
Comment (chuaindia) [reply rental.rb]: Done.

File: person.rb (hunk 2 above)
Comment (chuaindia) [reply person.rb]: Method add_rentals added.

File: book.rb (hunk 1 above)
Comment (chuaindia) [reply book.rb]: Method add_rentals added.

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @chuaindia, 

To highlight 👍🏽 
- You created getters and setters methods
- You created a method to add rentals

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Comment by chuaindia** [comment]
Thank You very much.

**Pull request merged**